pkg/controller/clusters/workers.go

```diff
@@ -46,11 +47,14 @@ type WorkerGroup struct {
 }
 
 func (g *WorkerGroup) objects() []client.Object {
-	return []client.Object{
-		g.KubeadmConfigTemplate,
-		g.MachineDeployment,
-		g.ProviderMachineTemplate,
+	objs := make([]client.Object, 0, 3)
+	objs = append(objs, g.KubeadmConfigTemplate, g.MachineDeployment)
+
+	if !reflect.ValueOf(g.ProviderMachineTemplate).IsNil() {
```

Comment:
is this basically checking `g.ProviderMachineTemplate != nil`?

Reply:
yes but ProviderMachineTemplate is not a pointer and the `g.ProviderMachineTemplate != nil` check fails causing nil panics when we tried it.